tests/write-json.test.js

```diff
@@ -0,0 +1,91 @@
+const { promises: fs } = require('fs');
+const { convertToJson } = require('../scripts/utils');
+const writeJSON = require("../scripts/utils/write-json");
+const {yamlString, jsonObject} = require("./fixtures/utilsData");
+
+jest.mock('fs', () => ({
+  promises: {
+    readFile: jest.fn(),
+    writeFile: jest.fn(),
+  },
+}));
+
+jest.mock('../scripts/utils', () => ({
+  convertToJson: jest.fn(),
+}));
+
+describe('writeJSON', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  test('should read a file, convert it to JSON, and write the JSON to another file', async () => {
+    const readPath = 'config/testInput.yaml';
+    const writePath = 'config/testOutput.json';
+
+    fs.readFile.mockResolvedValue(yamlString);
+    convertToJson.mockReturnValue(jsonObject);
+
+    await writeJSON(readPath, writePath);
+
+    expect(fs.readFile).toHaveBeenCalledWith(readPath, 'utf-8');
+    expect(convertToJson).toHaveBeenCalledWith(yamlString);
+    expect(fs.writeFile).toHaveBeenCalledWith(writePath, JSON.stringify(jsonObject));
+  });
+
+  test('should log an error and throw if reading the file fails', async () => {
+    const readPath = 'config/testInput.yaml';
+    const writePath = 'config/testOutput.json';
+    const error = new Error('File read error');
+
+    fs.readFile.mockRejectedValue(error);
+    const consoleErrorSpy = jest.spyOn(console, 'error').mockImplementation(() => {});
```

Comment:
Why do we need to `mockImplementation` for console.error function?

Reply:
to prevent the output to log in the console during tests. If we log errors to the console that can clutter the output and make it harder to identify test failures when we have multiple tests

Reply:
I don't think there will be any problem with console error messages. You shouldn't mock this implementation as it will help us to detect the flow of the function, on a desired test.

Reply:
Okay 👍 I'll update it